packages/core/src/providers/starknet/manager.ts

```diff
@@ -69,8 +69,8 @@ export function useStarknetManager(): StarknetState {
       const [account] = await window.starknet.enable()
       dispatch({ type: 'set_account', account })
       const starknet = getStarknet()
-      if (starknet.signer) {
-        dispatch({ type: 'set_provider', provider: starknet.signer })
+      if (starknet.account) {
+        dispatch({ type: 'set_provider', provider: starknet.account })
```

Comment:
hm im not sure i understand how this is working. does the starknet account implement `ProviderInterface`?

Reply:
It does. `AccountInterface` is a subclass of `ProviderInterface`